src/CrossScopeValidator.ts

```diff
@@ -274,10 +274,11 @@ export class CrossScopeValidator {
     }
 
     getProvidedTree(scope: Scope) {
-        if (this.providedTreeMap.has(scope)) {
-            return this.providedTreeMap.get(scope);
+        const lowerScopeName = scope.name?.toLowerCase();
```

Comment:
Just thinking about performance for a moment, could we avoid calling .toLowerCase on the scope name? I think it's already all lower case, but if it's not, then we could create a new field on scope, and just use that. Could cut down on some garbage collection churn?